test/VaultOUSDLvUSD.js

```diff
@@ -0,0 +1,172 @@
+        let mockOUSDContract = await ethers.getContractFactory("OUSDMockToken");
+        let lvUSDContract = await ethers.getContractFactory("LvUSDToken");
+        [owner, addr1, addr2, ...addrs] = await ethers.getSigners();
+        OUSDMockToken = await mockOUSDContract.deploy();
+        lvUSDToken = await lvUSDContract.deploy()
+        vaultToken = await contract.deploy(OUSDMockToken.address, lvUSDToken.address);
+
+        // Mint initial amount on OUSD token, will be used by all tests/
+        // "owner" is calling this mint function. addr1 is our client's address
+        await OUSDMockToken.mint(addr1.address, defaultUserOUSDBalance)
+        await lvUSDToken.mint(addr1.address, defaultUserLvUSDBalance)
+    });
+
+    async function depositOUSDFundsInVault(amount) {
+        balance = await vaultToken.getVaultOUSDBalance();
+        await depositOUSDFundsInVaultFromAccount(addr1, amount)
+        expect(await vaultToken.getVaultOUSDBalance()).to.equal(balance + amount)
+    };
+
+    async function depositOUSDFundsInVaultFromAccount(fromAccount, amount) {
+        // first set it to 0 to mitigate potential abuse https://ethereum.org/en/developers/tutorials/erc20-annotated-code/
+        await OUSDMockToken.connect(fromAccount).approve(vaultToken.address, 0);
+        await OUSDMockToken.connect(fromAccount).approve(vaultToken.address, amount);
+        // console.log("test:Vault:depositFunds:addr1 allowance %s", await)
+        await vaultToken.depositOUSD(fromAccount.address, amount);
+    };
+
+    async function checkVaultOUSDBalance(expected) {
+        let vaultBalance = await vaultToken.getVaultOUSDBalance();
+        expect(vaultBalance).to.equal(expected);
+        expect(await OUSDMockToken.balanceOf(vaultToken.address)).to.equal(expected)
+    };
+
+    async function depositLvUSDIntoVault(amount) {
+        await lvUSDToken.connect(addr1).approve(vaultToken.address, 0);
+        await lvUSDToken.connect(addr1).approve(vaultToken.address, amount);
+        await vaultToken.depositLvUSD(addr1.address, amount);
+    }
+
+    async function checkVaultLvUSDBalance(expected) {
+        expect(await lvUSDToken.balanceOf(vaultToken.address)).to.equal(expected)
+        expect(await vaultToken.getVaultLvUSDBalance()).to.equal(expected);
+
+    }
+
+    describe("Deposit OUSD funds", function () {
+        it("Should be able to deposit OUSD into vault", async function () {
+            let depositedAmount = 1000;
+            console.log("test:Vault:depositFunds:contract address", vaultToken.address);
+            await depositOUSDFundsInVault(depositedAmount)
+            await checkVaultOUSDBalance(depositedAmount)
+        });
+
+        it("Should revert with transfer amount exceeds balance when trying to transfer more then client owns", async function () {
+            // getting current user balance and adding 1 (to exceed balance and fail)
+            let depositedAmount = await OUSDMockToken.balanceOf(addr1.address) + 1;
```

Comment:
nitpicking - depositedAmount --> depositedOUSDAmount

Also the `+1` I think we should acknowledge decimal (for example - add a comment like "this 1 is actually 1 / 10^18")